lib/apipie/rspec/response_validation_helper.rb

```diff
@@ -89,18 +89,19 @@ module Apipie::ControllerValidationHelpers
     # this method is injected into ActionController::Base in order to
     # get access to the names of the current controller, current action, as well as to the response
     def schema_validation_errors_for_response
-      unprocessed_schema = Apipie::json_schema_for_method_response(controller_name, action_name, response.code, true)
+      resource_name = Apipie::app.get_resource_name(self.class)
```

Comment:
Is `Apipie::app.get_resource_name` able to generate unique resource names just based on the `self.class` that gets passed in? Was wondering if there's a chance for the `resource_name` generated by a particular controller action to collide with a different action from the same class?

Reply:
The resource name only describes the controller. There is logic in `get_resource_name` to use a prefixed name for namespaced controllers (in our case controllers under `Admin` or `AdUser`). This is enough to prevent conflicts in our autoloading setup (since class names have to match file paths).

https://github.com/instacart/apipie-rails/blob/c7684334a3b395974be34f7c778e3b9604953a12/lib/apipie/application.rb#L371-L386

You can also override the resource name by defining a `resource_name` in the `resource_description` block on the controller.